Another round of fixes

DisasterMoo opened the pull request:
- Fixed #1216 by checking for server side
- Fixed a minor issue in TOP tooltips
- Fixed #1219 by adding a missing `copy()` stack item small vessel firing recipe (fire pit shrink stacks)
- Fixed #1225 by allowing TFC chests to be used
- Fixed #1231
- Fixed #1232 by making maizes follow other grains
- Fixed #1228 by adding a new `IIngredient` implementation that checks for not rotten foods
- Fixed #1220 by checking if brining or picking food has the trait already
- May have Fixed #1229 by making server side update hitboxes once per day / when animal eats.
- Added more info to fruit trees and berry bushes (Closes #1211)
- Fixed #1093 by changing conflicting function signatures between `IEntityStandAttack` and vanilla's polar bear.
- Config reorganization (Closes #515)

DisasterMoo committed:
Fixed #1216.

DisasterMoo committed:
Fixed a minor issue, since TOP tooltips are called server side.

DisasterMoo committed:
Fixed #1219.

DisasterMoo committed:
Entities that can be chested now accepts TFC chests (Closes #1225).

DisasterMoo committed:
Fixed the flying baby goat (Closes #1231).

DisasterMoo committed:
Fixed #1232.

DisasterMoo committed:
Fixed rotten foods being accepted as valid ingredients on recipes (Closes #1228).

DisasterMoo committed:
Fixed #1220.

DisasterMoo committed:
Animals' hitboxes are now updated once per day / whenever it eats food (may have fixed #1229).

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/blocks/devices/BlockFirePit.java:
```diff
@@ -274,25 +274,14 @@ else if (tile.getCookingPotStage() == TEFirePit.CookingPotStage.FINISHED)
         return true;
     }
 
-    @Override
-    public void onEntityWalk(World worldIn, BlockPos pos, Entity entityIn)
-    {
-        IBlockState state = worldIn.getBlockState(pos);
-        if (state.getValue(LIT) && !entityIn.isImmuneToFire() && entityIn instanceof EntityLivingBase)
-        {
-            entityIn.attackEntityFrom(DamageSource.IN_FIRE, 1.0F);
-        }
-        super.onEntityWalk(worldIn, pos, entityIn);
-    }
-
```
This was intentionally added to fix #1025, which the result behavior was that fire pits only damaged you in very select circumstances. Is the problem not the below change to the wrong `super` call?

DisasterMoo replied:
Oh, no that wrong super call did nothing. Looking at the call chain the default implementation of both methods does nothing. Now, removing `#onEntityWalk` fixed. 

Looking at the call chain, `#onEntityWalk` is called from `Entity#move()L1039` when said entity is in the block or one block above it if the above block is empty/AIR, so, if there is no other way to fix #1025 we're gonna have to choose which one stays fixed.

DisasterMoo replied:
Removed fix as per discord discussion

DisasterMoo committed:
Added missing fertilized tooltip to oviparous animals.

DisasterMoo committed:
Some improvement over berry bush and fruit trees waila information (Closes #1211):
- Both now show fruit growth when on fruiting season
- Fruit trees now shows trunk growth

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/blocks/agriculture/BlockFruitTreeTrunk.java:
```diff
@@ -58,7 +58,7 @@ public static BlockFruitTreeTrunk get(IFruitTree tree)
     }
 
-    private final IFruitTree tree;
+    public final IFruitTree tree;
```
Hey. Where be my `getFruitTree` with private behind the scenes fields!

DisasterMoo replied:
Fine fine, I also thought on doing it properly, but.... this means i have to do the same thing in 99999 other classes that also has this inconsistency 😨 

(numbers may be a jest, but there are some other blocks that have a public property field that should be private with a getter)

Welp, time to mark this PR as a draft and start refactoring!

alcatrazEscapee commented on src/main/java/net/dries007/tfc/objects/te/TETickCounter.java:
```diff
@@ -25,12 +25,14 @@ public long getTicksSinceUpdate()
     public void resetCounter()
     {
         lastUpdateTick = CalendarTFC.PLAYER_TIME.getTicks();
+        markForSync();
         markDirty();
```
`markForSync` already calls `markDirty`, no need to call both here. Same as below

DisasterMoo committed:
Large Config Reorganization (Closes #515):
- Fixed Typo in female polar bear lang key (Closes #1251).
- Hyenas respawns in groups of 3 to 6 (Closes #1250).
- Changed how crops, berry bushes and fruit trees generation to be configurable.
- Added time modifiers for crops, berry bushes and fruit trees.
- Added various config options to animals.

alcatrazEscapee commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.RangeInt(min = 1, max = 10_000)
+            public int richOreMetalAmount = 35;
+
+            @Config.Comment("Enable ingot pile placement in world.")
+            @Config.LangKey("config." + MOD_ID + ".general.misc.enableIngotPiles")
+            public boolean enableIngotPiles = true;
```
Should be in overrides, as it's a binary enable/disable of a feature.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.player.inventoryCraftingMode")
+            public InventoryCraftingMode inventoryCraftingMode = InventoryCraftingMode.ENABLED;
+
+            @Config.Comment("Minimum health modifier player can obtain with low nutrition.")
```
Include math. Example: Player health * this number = minimum health

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.overrides.removeVanillaLoots")
+            public boolean removeVanillaLoots = true;
+
+            @Config.Comment({"If true, this will force the gamerule naturalRegeneration to be false. ", "Note: this DOES NOT AFFECT TFC's natural regeneration. If you set naturalRegeneration to true, then you will have both TFC regeneration and normal vanilla regeneration (which is much faster)"})
```
This should have an enter in it at the end of the comment line so it's clear it's two lines.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.tree.enableSaplings")
+            public boolean enableSaplings = true;
+
+            @Config.Comment("Log return rate of stone axes (eg: How efficiently it is)")
```
efficient not efficiently
```suggestion
            @Config.Comment("Chance per log for an item to drop when using a stone axe.")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.tree.leafStickDropChanceBonus")
+            public double leafStickDropChanceBonus = 0.25;
+
+            @Config.Comment("Bonus leaf drop stick chance for sticks tool classes")
```
Should include info on how to find these.

DisasterMoo replied:
You mean this?
```java
            @Config.Comment("Chance that leaves will drop more sticks when harvested with configured tool classes.")
            @Config.RangeDouble(min = 0, max = 1)
            @Config.LangKey("config." + MOD_ID + ".general.tree.leafStickDropChanceBonus")
            public double leafStickDropChanceBonus = 0.25;

            @Config.Comment("Tool classes that have the configured bonus to drop more sticks when harvesting leaves.")
            @Config.LangKey("config." + MOD_ID + ".general.tree.leafStickDropChanceBonusClasses")
            public String[] leafStickDropChanceBonusClasses = new String[] {"knife", "scythe"};
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+        {
+            @Config.Comment("Should living in a chunk block hostile mob and predators spawning over time?")
+            @Config.LangKey("config." + MOD_ID + ".general.spawn_protection.enable")
+            public boolean enable = true;
```
Should likely split this in two. One for Hostiles and one for predators.
This ties into the Y check option as it would allow people to set the Y to zero and only block underground mobs if they wish.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+        public static final class DamageCFG
+        {
+            @Config.Comment("Damage Source Types that will default to Slashing damage.")
```
How to get them? This goes for all of the following areas too. If we don't have a tool built into TFC to get the required entries something should be added and explained.

DisasterMoo replied:
What do you mean? I don't think we should add a a `/CT mainhand` like command for something like `DamageSource`, same thing for `Entity` entries, they are all Registry entries that already are in forge and vanilla, and other utility mods can dump them if a pack maker so requires.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.damage.slashingSources")
+            public String[] slashingSources = new String[] {};
+
+            @Config.Comment("Damage Source Types that will default to Piercing damage.")
```
How to get them?

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.damage.piercingSources")
+            public String[] piercingSources = new String[] {"arrow", "cactus", "thorns"};
+
+            @Config.Comment("Damage Source Types that will default to Crushing damage.")
```
How to get them?

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.damage.crushingSources")
+            public String[] crushingSources = new String[] {"anvil", "falling_block"};
+
+            @Config.Comment("Damage Source Entities that will default to Slashing damage.")
```
How to get them?

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.damage.slashingEntities")
+            public String[] slashingEntities = new String[] {"minecraft:wither_skeleton", "minecraft:vex", "minecraft:vindication_illager", "minecraft:zombie_pigman", "minecraft:wolf", "minecraft:polar_bear"};
+
+            @Config.Comment("Damage Source Entities that will default to Piercing damage.")
```
How to get them?

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.damage.piercingEntities")
+            public String[] piercingEntities = new String[] {"minecraft:stray", "minecraft:skeleton"};
+
+            @Config.Comment("Damage Source Entities that will default to Crushing damage.")
```
How to get them?

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.player.minHealthModifier")
+            public double minHealthModifier = 0.2d;
+
+            @Config.Comment("Maximum health modifier player can obtain with high nutrition.")
```
Include math. Example: Player health * this number = total health.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.player.maxHealthModifier")
+            public double maxHealthModifier = 3d;
+
+            @Config.Comment("Modifier for how quickly the players becomes thirsty.")
```
Include comparison values. What is 8? 8x? 8 x another number? How does it compare to vanilla hunger for example.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.player.thirstModifier")
+            public double thirstModifier = 8.0;
+
+            @Config.Comment("Modifier for how quickly the player will naturally regenerate health.")
```
Include comparison values. Is 1 vanilla?

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.player.peacefulDifficultyPassiveRegeneration")
+            public boolean peacefulDifficultyPassiveRegeneration = false;
+
+            @Config.Comment("A multiplier for passive exhaustion (how fast your hunger decays. 0 = hunger doesn't decay passively, higher values = faster decay).")
```
Include comparison values. Is this a multiplier on vanilla values?

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.RangeDouble(min = 0, max = 100)
+            public double passiveExhaustionMultiplier = 1;
+
+            @Config.Comment("This is the amount of hunger (consumed) required for a player to completely refresh their nutrition stats. In TFC, almost all foods have 4 hunger, so this is 4x [number of foods required to refresh nutrition stats]. This will update next time you consume a food.")
```
This is confusing.
```suggestion
            @Config.Comment(
"The amount of replenished hunger before the player's nutrition will lose the first food consumed. Most TFC foods have 4 hunger.", 
"Example: Multiply Vanilla hunger(20) by 4 to get one food bars worth of food before the first food is lost from the cycle." )
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+        public static final class WorldCFG
+        {
+            @Config.RequiresMcRestart
+            @Config.Comment({"This controls how the temperature gradient appears near the equator.", "1: south of equator is hot, north of equator is cold", "-1: south of equator is cold, north of equator is hot"})
```
```suggestion
            @Config.Comment("Sets temperature in relation to the equator change. North = Cold, South = Hot or North = Hot, South = Cold")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.world.temperatureMode")
+            public TemperatureMode temperatureMode = TemperatureMode.CYCLIC;
+
+            @Config.Comment({"This controls the size of the temperature regions. The size of each temperature zone is determined by a sin wave. This represents the period of the wave."})
```
Perhaps a better explanation could be done here for people who don't understand sine waves?

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.world.looseRocksFrequency")
+            public int looseRocksFrequency = 18;
+
+            @Config.Comment("This controls how deep loose rocks scans for veins when generating. Higher values = more ore samples.")
```
```suggestion
            @Config.Comment("This is how deep in blocks from the surface a loose rock will scan for a vein when generating, Higher values = More veins spawn samples thus adding more samples.")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+            @Config.RequiresMcRestart
+            @Config.RangeDouble(min = 0.05, max = 0.4)
+            @Config.Comment("This controls how spread the rainfall distribution is. Higher values mean the world will be distributed towards the extremes more. WARNING: This is can cause very weird world generation conditions.")
```
This comment should be improved. What are the "extremes"?
I don't know enough about it to correct it.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+            @Config.RequiresMcRestart
+            @Config.RangeDouble(min = 0.05, max = 0.4)
+            @Config.Comment("This controls how spread the flora diversity distribution is. Higher values mean the world will be distributed towards the extremes more. WARNING: This is can cause very weird world generation conditions.")
```
This comment should be improved. What are the "extremes"?
I don't know enough about it to correct it.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+            @Config.RequiresMcRestart
+            @Config.RangeDouble(min = 0.05, max = 0.4)
+            @Config.Comment("This controls how spread the flora density distribution is. Higher values mean the world will be distributed towards the extremes more. WARNING: This is can cause very weird world generation conditions.")
```
This comment should be improved. What are the "extremes"?
I don't know enough about it to correct it.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+            @Config.RequiresMcRestart
+            @Config.Comment({"This controls which registered entities can respawn in TFC biomes.",
+                "You must specify by following the pattern 'modid:entity <rarity> <minGroupSpawn> <maxGroupSpawn>'",
```
How to get entity ID?
What conditions does respawn use?
If it fails there should be a logline generated and an example of it added here.

DisasterMoo replied:
This falls in the same category as the changes you requested for damages:
- **How to get entity ID?** Use utility mods, it's not a secret that only TFC know.
- **What conditions does respawn use?** Exactly the same as vanilla: randomly selected from the list of creatures.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+        public static final class WorldRegenCFG
+        {
+            @Config.Comment("The minimum time for a chunk to be unoccupied for it's resources to begin to naturally regenerate. (In days). After this amount, regeneration will scale up based on how long since this duration, up to a maximum of 4x")
```
What resources? Currently this sounds like the whole chunk regenerates slowly over time.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.berryBushMovementModifier")
+            public double berryBushMovementModifier = 0.1;
+
+            @Config.Comment("Generic snow movement modifier. Lower = Slower, Higher = Faster. 1 = No slow down.")
```
Add math to match leafs and bush.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.defaultMonthLength")
+            public int defaultMonthLength = 8;
+
+            @Config.Comment("Percentage chance that plants (not crops) grow each update. Smaller number = slower growth.")
```
This needs to be explained better.
```suggestion
            @Config.Comment("Chance for a plant to grow each random tick, does not include crops. Lower = slower growth."")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.minimumPlantMovementModifier")
+            public double minimumPlantMovementModifier = 0;
+
+            @Config.Comment("Drop chance for gem from raw stone")
```
This should include how to find the chance per grade of gem if that is also factored in.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.dictionaryPlates")
+            public boolean dictionaryPlates = false;
+
+            @Config.Comment("List of fluids allowed to be picked up by wooden bucket")
```
How to find the options?

DisasterMoo replied:
Same as the others, use utility mods.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.woodenBucketWhitelist")
+            public String[] woodenBucketWhitelist = new String[] {"fresh_water", "hot_water", "salt_water", "water", "limewater", "tannin", "olive_oil", "vinegar", "rum", "beer", "whiskey", "rye_whiskey", "corn_whiskey", "sake", "vodka", "cider", "brine", "milk", "milk_curdled", "milk_vinegar", "white_dye", "orange_dye", "magenta_dye", "light_blue_dye", "yellow_dye", "lime_dye", "pink_dye", "gray ...
+
+            @Config.Comment("List of fluids allowed to be picked up by blue steel bucket")
```
How to find the options?

DisasterMoo replied:
Same as the others, use utility mods.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.enableIngotPiles")
+            public boolean enableIngotPiles = true;
+
+            @Config.Comment("Add iron ore dictionary to wrought iron?")
```
This should include the tags it adds.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.dictionaryIron")
+            public boolean dictionaryIron = false;
+
+            @Config.Comment("Add plate ore dictionary to sheets?")
```
This should include the tags it adds.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.blueSteelBucketWhitelist")
+            public String[] blueSteelBucketWhitelist = new String[] {"lava"};
+
+            @Config.Comment("List of fluids allowed to be picked up by red steel bucket")
```
How to find the options?

DisasterMoo replied:
Same as the others, use utility mods.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".general.misc.redSteelBucketWhitelist")
+            public String[] redSteelBucketWhitelist = new String[] {"fresh_water", "hot_water", "salt_water", "water"};
+
+            @Config.Comment("Entities that can be plucked for feathers.")
```
(I made this one and still screwed it up)
How to find the names?

DisasterMoo replied:
Same as the others, use utility mods.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+        public static final class BlastFurnaceCFG
+        {
+            @Config.Comment("How fast the blast furnace consume fuels (compared to the charcoal forge).")
```
Needs maths.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".devices.lamp.burnRate")
+            public double burnRate = 0.125;
+
+            @Config.Comment("Which fluids are valid fuels for lamps?")
```
How to get values?

DisasterMoo replied:
Same as the others, use utility mods.

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+        public static final class AlpacaCFG
+        {
+            @Config.Comment("How many days it is needed for this animal to be considered a fully grown adult?")
```
I'm only going to do this one animal bit. If you want to change them, copy paste to the rest, too many to mark.

```suggestion
            @Config.Comment("How many days until this animal is a full grown adult?")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".animals.elder")
+            public int elder = 768;
+
+            @Config.Comment("How many days it is needed for this animal finish a gestation period?")
```
```suggestion
            @Config.Comment("How many days until this animal gives birth?")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".animals.gestation")
+            public int gestation = 59;
+
+            @Config.Comment("How many babies are born when this animal finishes gestation?")
```
```suggestion
            @Config.Comment("How many babies are born when this animal gives birth?")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".animals.adulthood")
+            public int adulthood = 192;
+
+            @Config.Comment("How many more days it is needed for this animal to be considered an elder? 0 = Disable")
```
```suggestion
            @Config.Comment("How many days after becoming an adult until this animal is old? 0 = Disable")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".animals.babies")
+            public int babies = 1;
+
+            @Config.Comment("How many ticks it is needed for this animal grow back wool?")
```
```suggestion
            @Config.Comment("How many ticks are needed for this animal grow back wool?")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".animals.woolTicks")
+            public int woolTicks = 96_000;
+
+            @Config.Comment("If this animal is an elder, what is the chance it is gonna die on every new day? 0 = Disable")
```
```suggestion
            @Config.Comment("Chance that old animals will die at the start of a new day. 0 = Disable")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+
+        public static final class TemperatureCFG
+        {
+            @Config.Comment("Modifier for how quickly items gains and loses heat. Smaller number = slower temperature changes.")
```
```suggestion
            @Config.Comment("Modifier for how quickly items will gain or lose heat. Smaller number = slower temperature changes.")
```

Claycorp commented on src/main/java/net/dries007/tfc/ConfigTFC.java:
```diff
@@ -46,614 +31,1423 @@ public static void onConfigChangedEvent(ConfigChangedEvent.OnConfigChangedEvent
+            @Config.LangKey("config." + MOD_ID + ".devices.temperature.globalModifier")
+            public double globalModifier = 0.5;
+
+            @Config.Comment("Modifier for how quickly devices (i.e. charcoal forge, fire pit) gain and lose heat. Smaller number = slower temperature changes.")
```
```suggestion
            @Config.Comment("Modifier for how quickly devices (i.e. charcoal forge, fire pit) will gain or lose heat. Smaller number = slower temperature changes.")
```

Claycorp reviewed (commented):
Just a bunch of random changes. Seems good enough otherwise for the rest of 1.12

DisasterMoo committed:
Changes requested.
